app/models/book.rb

```diff
@@ -0,0 +1,26 @@
+class Book < ApplicationRecord
+  belongs_to :category
+  has_many :markers, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+
+  validate :min_publish_date, :max_publish_date
+  validates :title, length: {maximum: Settings.string_maximum}, presence: true
+  validates :description, presence: true
+  validates :number_pages, presence: true
+  validates :price, presence: true
+  validates :author, length: {maximum: Settings.name_maximum}, presence: true
+
+  private
+
+  def min_publish_date
+    return unless publish_date <= Settings.min_date
```

Comment:
c nghĩ k cần thiết phải giới hạn min_date đâu hề?
n xuất bản năm nào là quyền của nó chứ?
sách 80 năm trước em k cho phép n đọc à?

Reply:
Em cũng định hỏi chị đó. Khi mà nhập date là 1 nó sẽ có exeption là : mysql2 rejected. !

Reply:
date là 1?? what's the hell!!!!!!!! date là dạng ngày tháng năm mà?
1 thì sai định dạng là đúng rồi còn kêu ca gì?

Reply:
@DuongNT1512 Ý em là 01/01/100 nó sẽ quá min của mysql2 và exeption ý ạ

Reply:
thì validate n khoảng 100 năm đổ lại đây. k để quá lâu như vậy nữa

Reply:
@DuongNT1512 Thì min_date em đang để value là 100 trở lạ đây đó mà chị iu ;p

Reply:
À ý chị vẫn là 100 years ago đó hả :v: